src/target/riscv/riscv.c

```diff
@@ -4499,55 +4509,47 @@ COMMAND_HANDLER(riscv_exec_progbuf)
 	return ERROR_OK;
 }
 
-COMMAND_HANDLER(riscv_set_enable_eq_match_trigger)
+COMMAND_HANDLER(riscv_set_enable_trigger_type)
 {
 	struct target *target = get_current_target(CMD_CTX);
 	RISCV_INFO(r);
 
-	if (CMD_ARGC == 0) {
-		command_print(CMD, "equality match trigger enabled: %s", r->enable_equality_match_trigger ? "on" : "off");
-		return ERROR_OK;
-	} else if (CMD_ARGC == 1) {
-		COMMAND_PARSE_ON_OFF(CMD_ARGV[0], r->enable_equality_match_trigger);
-		return ERROR_OK;
-	}
-
-	LOG_ERROR("Command takes 0 or 1 parameters");
-	return ERROR_COMMAND_SYNTAX_ERROR;
-}
+	if (CMD_ARGC == 2) {
+		bool enable_for_wp = false;
 
-COMMAND_HANDLER(riscv_set_enable_napot_trigger)
-{
-	struct target *target = get_current_target(CMD_CTX);
-	RISCV_INFO(r);
+		if (!strcmp(CMD_ARGV[1], "wp"))
+			enable_for_wp = true;
+		else if (!strcmp(CMD_ARGV[1], "none"))
+			enable_for_wp = false;
+		else
+			return ERROR_COMMAND_SYNTAX_ERROR;
 
-	if (CMD_ARGC == 0) {
-		command_print(CMD, "NAPOT trigger enabled: %s", r->enable_napot_trigger ? "on" : "off");
-		return ERROR_OK;
-	} else if (CMD_ARGC == 1) {
-		COMMAND_PARSE_ON_OFF(CMD_ARGV[0], r->enable_napot_trigger);
-		return ERROR_OK;
+		if (!strcmp(CMD_ARGV[0], "all")) {
+			r->wp_enable_equality_match_trigger = enable_for_wp;
+			r->wp_enable_napot_trigger = enable_for_wp;
+			r->wp_enable_ge_lt_trigger = enable_for_wp;
+		} else if (!strcmp(CMD_ARGV[0], "eq")) {
+			r->wp_enable_equality_match_trigger = enable_for_wp;
+		} else if (!strcmp(CMD_ARGV[0], "napot")) {
+			r->wp_enable_napot_trigger = enable_for_wp;
+		} else if (!strcmp(CMD_ARGV[0], "ge_lt")) {
+			r->wp_enable_ge_lt_trigger = enable_for_wp;
+		} else {
+			return ERROR_COMMAND_SYNTAX_ERROR;
+		}
+	} else if (CMD_ARGC != 0) {
+		return ERROR_COMMAND_SYNTAX_ERROR;
 	}
 
-	LOG_ERROR("Command takes 0 or 1 parameters");
-	return ERROR_COMMAND_SYNTAX_ERROR;
-}
-
-COMMAND_HANDLER(riscv_set_enable_ge_lt_trigger)
-{
-	struct target *target = get_current_target(CMD_CTX);
-	RISCV_INFO(r);
-
-	if (CMD_ARGC == 0) {
-		command_print(CMD, "ge-lt triggers enabled: %s", r->enable_ge_lt_trigger ? "on" : "off");
-		return ERROR_OK;
-	} else if (CMD_ARGC == 1) {
-		COMMAND_PARSE_ON_OFF(CMD_ARGV[0], r->enable_ge_lt_trigger);
-		return ERROR_OK;
-	}
+	command_print(CMD, "Triggers enabled:\n"
```

Comment:
I'd recommend to call this "trigger feature configuration" or similar.

Instead of `on` / `off`, I'd recommend to print `enabled` / `disabled` for maximum user clarity.

```suggestion
	command_print(CMD, "Trigger feature configuration:\n"
```

Reply:
Addressed.